### ISSUE TITLE
Story Block: accessibility inspector becomes unusable after an image is uploaded to the Story block.

### ISSUE THREAD

**Issue by worldomonation** [Automattic/jetpack#32976]
### Impacted plugin

Jetpack

### Quick summary

When using the Story block, uploading an image "bugs out" the accessibility inspector.

This may not be an issue for most sighted users, but for those relying on VoiceOver, as well as E2E tests (which rely on accessible locators), this makes the platform unusable.

### Steps to reproduce

1. start a new post.
2. add the story block.
3. enter F12 or right click on block > inspect (on Firefox, you can directly open the accessibility inspector by right click on block > accessibility inspector).
4. view the accessibility inspector.
5. upload an image to the Story block.
6. attempt to use the accessibility inspector.

### A clear and concise description of what you expected to happen.

Accessibility inspector remains usable (eg. like with any other block).

### What actually happened

Accessibility inspector "bugs" out and stops rendering the DOM with accessible locators.

For example, Playwright tests execute and locate elements based on the accessible locators. 
Publishing a post with a Story block containing at least one image will fail tests because when accessing the published post, the `getByRole('main')` method fails to locate any elements - something never observed with any other blocks. Tests return to passing if the Story block is not added to the published post.

https://github.com/Automattic/jetpack/assets/6549265/63ae5ab9-8db6-4107-a37d-41645c3655d5

### Impact

All

### Available workarounds?

Yes, difficult to implement

**Comment by cuemarie** [comment]
📌 REPRODUCTION RESULTS
- Tested on Simple – **Replicated** / Could Not Replicate / Uncertain
- Tested on Atomic – **Replicated** / Could Not Replicate / Uncertain
- Tested on P2 (optional) – Replicated / Could Not Replicate / Uncertain
- Replicable outside of Dotcom – Yes/No

📌 FINDINGS/SCREENSHOTS/VIDEO
Replicated as described:

https://github.com/Automattic/jetpack/assets/27249804/6b6ddbb4-2b01-4aab-a882-b5284d656a82
